src/output_mqtt.c

```diff
@@ -125,6 +132,11 @@ static mqtt_client_t *mqtt_client_init(struct mg_mgr *mgr, char const *host, cha
         exit(1);
     }
 
+    if (will_topic != NULL) {
+        ctx->message_id++;
+        mg_mqtt_publish(ctx->conn, will_topic, ctx->message_id, MG_MQTT_QOS(0) | MG_MQTT_RETAIN, mqtt_lwt_online, strlen(mqtt_lwt_online));
+    }
+
```

Comment:
This doesn't work, because mg_set_protocol_mqtt isn't called yet, so this needs to happen after MG_EV_CONNECT

Reply:
I just commited a fix for that a few seconds ago. :)